util/travis/script.sh

```diff
@@ -3,21 +3,61 @@
 
 needs_compile || exit 0
 
+function perform_lint() {
+	CLANG_FORMAT=clang-format-3.9
+	${CLANG_FORMAT} --version
+
+	if [ "$TRAVIS_EVENT_TYPE" = "pull_request" ]; then
+		# Get list of every file modified in this pull request
+		files_to_lint="$(git diff --name-only --diff-filter=ACMRTUXB $TRAVIS_COMMIT_RANGE | grep '^src/[^.]*[.]\(cpp\|h\)$' || true)"
+	else
+		# Check everything for branch pushes
+		files_to_lint="$(find src/ -name '*.cpp' -or -name '*.h' | egrep -v "src/(gmp|lua|jsoncpp)")"
+	fi
+
+	for f in ${files_to_lint}; do
+		d=$(diff -u "$f" <(${CLANG_FORMAT} "$f") || true)
+		if ! [ -z "$d" ]; then
+			echo "$f not compliant to coding style, here is the fix:"
+			echo "$d"
+			fail=1
+		fi
+	done
+
+	if [ "$fail" = 1 ]; then
+		exit 0 # Not mandatory at this moment
+	fi
+
+	exit 0
+}
+
+if [ $(git diff --name-only --diff-filter=ACMRTUXB $TRAVIS_COMMIT_RANGE | grep -c '^src/[^.]*[.]\(c\|cpp\|h\)$') -eq 0 ]; then
+	exit 0
+fi
+
 if [[ $PLATFORM == "Unix" ]]; then
 	mkdir -p travisbuild
 	cd travisbuild || exit 1
+	if [[ "$LINT" == "1" ]]; then
+		# Lint with exit CI
```

Comment:
exact, it's another build, it doesn't need to be done on each build. When linting will be mandatory on some/all files it permits to know directly the problem is codestyle